rapids-cmake/cython-core/add_rpath_entries.cmake

```diff
@@ -0,0 +1,93 @@
+# =============================================================================
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cython_add_rpath_entries
```

Comment:
Can you go over the change of this being in `cython-core` instead of `cython`. Is the plan here to remove the `cython` function entirely?

Reply:
For the moment I've separated the two for prototyping purposes because I anticipated needing to make breaking changes in Cython. I was hoping to just create a new set of CMake files that would be mostly drop-in replacements, i.e. for a rapids-cmake consumer to switch to using scikit-build-core they would just change `include(rapids-cython)` to `include(rapids-cython-core)` and then the rest of the function calls would "just work". That would allow a piecemeal migration of RAPIDS libraries rather than introducing a breaking point. This seems especially important since we now have consumers outside of RAPIDS alone (e.g. legate libraries) and so providing a softer off ramp would be valuable.

Reply:
That makes sense. As we move out of draft we should encode this information somewhere in the docs so that people understand the differences between `rapids-cython` and `rapids-cython-core` and how you can easily switch between them.

We might also to think about how we can write some smoke tests that verify that both modules share a common subset of API

Reply:
Definitely, I'll keep that in mind. JFYI it'll probably be a while before this PR comes out of draft, there's a lot of upstream work around our CI workflows that needs to get resolved before I think we'll be ready for the scikit-build-core switch. Hopefully 23.10.